packages/main/src/plugin/proxy.ts

```diff
@@ -22,6 +22,30 @@ import { Emitter } from './events/emitter.js';
 import { getProxyUrl } from './proxy-resolver.js';
 import { ProxyAgent } from 'undici';
 
+function ensureURL(urlstring: string | undefined): string | undefined {
+  if (urlstring) {
+    try {
+      const url = new URL(urlstring);
+      if (url.hostname) {
+        return urlstring;
+      }
+    } catch (err) {
+      /* empty */
+    }
+    return `http://${urlstring}`;
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function asURL(url: any): URL {
```

Comment:
It looks to me that url can only be URL, string or Request, no? Why using any?

Reply:
> It looks to me that url can only be URL, string or Request, no? Why using any?

It's related to the signature of the `fetch` method a few lines below

Reply:
but why we use any? we are overwriting the fetch function which accept only those three types 
![image](https://github.com/containers/podman-desktop/assets/49404737/2eb26458-4421-4b9d-8653-0ec8d72a4256)

Reply:
it's not a change made by this PR, just noticed it now but i think we can remove the any type

Reply:
> it's not a change made by this PR, just noticed it now but i think we can remove the any type

any was introduced because of the recent changes in Node types library

Reply:
mmm, saw it. The problem is that we update the types/node without updating the minimum node version and it creates discrepancy.
So, nevermind, nothing we can do here